deck.gl__geo-layers/index.d.ts

```diff
@@ -200,7 +200,7 @@ declare module '@deck.gl/geo-layers/tile-layer/tile-layer' {
       url: string;
       bbox: any;
       signal: AbortSignal;
-    }) => D[] | Promise<D[]> | null;
+    }) => D[] | null | Promise<D[] | null>;
```

Comment:
https://deck.gl/docs/api-reference/geo-layers/tile-layer#gettiledata

> It should return either the tile data or a Promise that resolves to the tile data.

>If `signal` is aborted, then throw or return **falsy** from `getTileData` so the data is not cached; do not return incomplete data. 

as the `getTileData` could be sync or async function, so when it's an async function and in the **falsy** scenario, the return type should be `Promise<null>`.